packages/events-api/src/adapter.ts

```diff
@@ -0,0 +1,156 @@
+import EventEmitter from 'events'; // tslint:disable-line
+import http, { IncomingMessage, ServerResponse } from 'http';
+import debugFactory from 'debug'; // tslint:disable-line
+import { createHTTPHandler } from './http-handler';
+
+const debug = debugFactory('@slack/events-api:adapter');
+
+/**
+ * An adapter for Slack's Events API.
+ */
+export class SlackEventAdapter extends EventEmitter {
+  /**
+   * The token used to authenticate signed requests from Slack's Events API.
+   */
+  public readonly signingSecret: string;
+
+  /**
+   * Whether to include the API event bodies in adapter event consumers.
+   */
+  public includeBody: boolean;
+
+  /**
+   * Whether to include request headers in adapter event consumers.
+   */
+  public includeHeaders: boolean;
+
+  /**
+   * When `true`, prevents the adapter from responding by itself and leaves that up to consumers.
+   */
+  public waitForResponse: boolean;
+
+  /**
+   * The HTTP server this adapter might be running, created in {@link start}.
+   */
+  private server?: http.Server;
+
+  /**
+   * @param signingSecret - The token used to authenticate signed requests from Slack's Events API.
+   * @param opts.includeBody - TODO:
+   * @param opts.includeHeaders - TODO:
+   * @param opts.waitForResponse - TODO:
+   */
+  constructor(
+    signingSecret: string | String,
+    {
+      includeBody = false,
+      includeHeaders = false,
+      waitForResponse = false,
+    }: EventAdapterOptions = {},
+  ) {
+    if (typeof signingSecret !== 'string' && !(signingSecret instanceof String)) {
```

Comment:
From a design perspective, I honestly don't know. But from the perspective of adding typings, because it already was 🤷‍♀ [(see above)](https://github.com/slackapi/node-slack-sdk/pull/799#discussion_r296864473).